sweep/sweeper.go

```diff
@@ -679,7 +722,7 @@ func (s *UtxoSweeper) bucketForFeeRate(
 // sweep fee rate, which is determined by calculating the average fee rate of
 // all inputs within that cluster.
 func (s *UtxoSweeper) clusterBySweepFeeRate() []inputCluster {
-	bucketInputs := make(map[int]pendingInputs)
+	bucketInputs := make(map[int]*bucketList)
```

Comment:
similar to above, pointer type is redundant for slices

Reply:
In this case, the slice is wrapped in a struct to allow it to be modified in-place using the `add` method. I tried to make `bucketList` just a type alias for the slice, but then you need to deal with the return value of `add` and also update the parent map that holds the (old) slice. I think it is easier to understand with the wrapping struct.

Reply:
gotcha that makes sense, i had forgotten about the member methods. thanks for the explanation